pkg/controller.v1/common/job.go

```diff
@@ -284,6 +284,11 @@ func (jc *JobController) ReconcileJobs(
 					if !match {
 						return fmt.Errorf("unable to recognize PodGroup: %v", klog.KObj(pg))
 					}
+
+					if q := volcanoPodGroup.Spec.Queue; len(q) > 0 {
+						queue = q
+					}
```

Comment:
@ckyuto This overriding looks fine, but this might be surprising for users since users can not observe this overriding anywhere.
So, could you implement validation webhooks here as well?

https://github.com/kubeflow/training-operator/tree/73398806a64899a36263860c1d34bdc2dd1ec291/pkg/webhooks

Reply:
Thanks @tenzen-y . I'm not sure what should I validate here. Can you elaborate?
I think this should be the expected behavior for most use cases. This change prevent the queue being overridden by the change of this [PR](https://github.com/kubeflow/common/pull/207), which originally just intended to sync minMember of queue name to replica but accidentally sync the queue name.

Reply:
@ckyuto You can prevent updating on queue name here: https://github.com/kubeflow/training-operator/blob/73398806a64899a36263860c1d34bdc2dd1ec291/pkg/webhooks/paddlepaddle/paddlepaddle_webhook.go#L61

Also, could you implement the same validation in the PyTorchJob, TFJob, and XGBoost as well?

Reply:
@ckyuto Ah, I guess that we can use CEL validation here instead of implementing validations like this.
Could you try it?

```
type SchedulingPolicy struct { 
 	MinAvailable           *int32                                 `json:"minAvailable,omitempty"` 
 	// +kubebuilder:validation:XValidation:rule="self == oldSelf", message="field is immutable"
 	Queue                  string                                 `json:"queue,omitempty"` 
[...]
```

https://github.com/kubeflow/training-operator/blob/73398806a64899a36263860c1d34bdc2dd1ec291/pkg/apis/kubeflow.org/v1/common_types.go#L226-L234

Reply:
No, none of above will work. These objects are validation for the value of kubeflow jobs, and these value won't be changed after the job is created. The queue value update happens on the queue in volcano podGroup.

Reply:
We should block updating the `.runPolicy.schedulingPolicy.queue` since the field is propagated to the PodGroup resource. We should not override or ignore the defined field without notification for the users.

I didn't want to say that we should introduce webhook validation instead of controller logic.
I wanted to say that we should implement the validation as well.

If we don't implement the webhook validations, users can not find the reason why updated `.runPolicy.schedulingPolicy.queue` was not propagated to the PodGroup resource.

Reply:
@ckyuto @tenzen-y Do we have any updates here ?
Do we need to include this fix as part of 1.8 release ?

Reply:
I'm still waiting for response from @ckyuto.

Reply:
I have an update. Please help review. Thanks!